integration_tests/src/main/python/data_gen.py

```diff
@@ -676,7 +676,27 @@ def start(self, rand):
         def make_dict():
             length = rand.randint(self._min_length, self._max_length)
             return {self._key_gen.gen(): self._value_gen.gen() for idx in range(0, length)}
-        self._start(rand, make_dict)
+        def make_dict_float():
+            # Make sure at most one key is nan
```

Comment:
Nit: could we mention more about the reason why at most one key is NaN?

e.g., 
```
In Spark, NaN = NaN returns true. At most one key is NaN to avoid duplicated key value.
This is different from python where NaN as dict key are not equal to each other, so it is possible to have multiple NaN as keys in one dict.
```

Reply:
Thanks for review, done.